apps/credential-showcase-api-server/src/database/repositories/PersonaRepository.ts

```diff
@@ -0,0 +1,59 @@
+import { eq } from 'drizzle-orm';
+import { Service } from 'typedi';
+import { DatabaseService } from '../../services/DatabaseService';
+import { NotFoundError } from '../../errors/NotFoundError';
+import { assets, personas } from '../schema';
+import { Persona, NewPersona, RepositoryDefinition } from '../../types';
+
+@Service()
+class PersonaRepository implements RepositoryDefinition<Persona, NewPersona> {
+    constructor(private readonly databaseService: DatabaseService) {}
+
+    async create(persona: NewPersona): Promise<Persona> {
+        const [result] = await (await this.databaseService.getConnection())
+            .insert(personas)
+            .values(persona)
+            .returning();
+
+        return result
+    }
+
+    async delete(id: string): Promise<void> {
+        await this.findById(id)
+        await (await this.databaseService.getConnection())
+            .delete(assets)
+            .where(eq(assets.id, id))
+    }
+
+    async update(id: string, persona: NewPersona): Promise<Persona> {
```

Comment:
we need to check if the images exists before adding the persona. we already have code for this that also returns a human readable error